src/nvim/tui/tui.c

```diff
@@ -283,7 +296,7 @@ static void terminfo_start(UI *ui)
     termname = xstrdup(term);
   }
   // Update 'term' option.
-  loop_schedule_deferred(&main_loop,
+  loop_schedule(&main_loop,
```

Comment:
you could just store termname in a variable instead, then there is no need to schedule anything.

Reply:
We are required to perform `set_tty_option("term")`, but there was no requirement of deferred schedule in TUI. One possible thing is to also store `termname` locally instead `LOOP_PROCESS_EVENTS` at the beginning of `tui_ui_client_init`.